numpy/distutils/unixccompiler.py

```diff
@@ -61,8 +61,9 @@ def UnixCCompiler__compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts
         raise CompileError(msg)
 
     # add commandline flags to dependency file
-    with open(obj + '.d', 'a') as f:
-        f.write(_commandline_dep_string(cc_args, extra_postargs, pp_opts))
+    if deps is not []:
```

Comment:
`if not deps:`
we want to check for empty list, not if it is that empty list (if that empty lists are singletons that is an implementational detail)